### ISSUE TITLE
Generating a correct Swagger definition with endpoints with multiple presents?

### ISSUE THREAD

**Issue by NickDiMucci** [ruby-grape/grape-swagger#776]
When I have an endpoint that does a single `present` (that returns a single grape-entity object in the response body), I can use grape-swagger to specify that this model entity is to be the response.

```
desc 'Return a customer', { success: API::Entities::Customer }
get do
  ...
  present customer
end
```
This will generate the following Swagger
```
"responses": {
          "200": {
            "description": "Show one particular customer",
            "schema": {
              "$ref": "#/definitions/CustomerEntity"
            }
          }
        },,
```

However, I also have several endpoints that do multiple `present`, so multiple entities can be returned. 
```
desc 'Return customer & their order'
get do
  ....
  present customer, with: API::Entities::Customer
  present order, with: API::Entities::Order
end
```

Or similarly, 

```
desc 'Return all customers'
get do
  ....
  present meta: { total_page: 100, per_page: 10 }
  present customers, with API::Entities::Customer
end
```

The issue is it seems grape-swagger only supports defining a single entity as the response entity, so I'm unable to generate a correct Swagger definition for the above endpoints. 

Is it possible to work around this, somehow? I'm not even quite sure if the OpenAPI 2 specification even supports doing such a thing; I haven't been able to find anything online about this.

The only solution I've been able to come up with is to essentially wrap the multiple `present`s into a single `present` using a wrapper entity

```
ret = {meta: create_meta_info(customers), customers: customers}
present ret, with: API::Entities::GetCustomersEntity
```

**Comment by jonmchan** [comment]
I would like to +1 the ask for this feature...

I would suggest that you can pass some sort of hash? In general, this should be sufficient for majority of use cases to fall in line with multiple `present`'s.

Proposal:

```
desc 'Return all customers',
   entity: { 
     customers: API::Entities::Customer,
     order: API::Entities::Order
   }
get do
  ....
  present order, with API::Entities::Order
  present customers, with API::Entities::Customer
end
```

An issue though would be how you can indicate if customer was a single record or within an array. Maybe we can create a helper for this that wraps the actual entity and indicates that it is an array?

```
desc 'Return all customers',
   entity: { 
     customers: entity_array(API::Entities::Customer),
     order: API::Entities::Order
   }
get do
  ....
  present order, with API::Entities::Order
  present customers, with API::Entities::Customer
end
```

**Comment by jonmchan** [comment]
https://github.com/ruby-grape/grape/issues/241 linking ruby grape issue that added multiple present. The same considerations to multiple present should be addressed in the grape-swagger implementation.

**Comment by LeFnord** [comment]
closed via #813
